src/components/tools/requests/Details.js

```diff
@@ -0,0 +1,184 @@
+        test_data_path: "",
+        source_url: "",
+        version: "",
+        documentation_url: "",
+        cmd_line: "",
+    });
+
+    const { isFetching: isDetailsFetching, error: detailsFetchError } =
+        useQuery({
+            queryKey: [ADMIN_TOOL_REQUEST_DETAILS_QUERY_KEY, { id: requestId }],
+            queryFn: getAdminToolRequestDetails,
+            config: {
+                enabled: requestId && open,
+                onSuccess: (resp) => {
+                    setDetails(resp);
+                },
+            },
+        });
+
+    return (
+        <Formik
+            initialValues={details}
+            enableReinitialize={true}
+        >
+            {({ handleSubmit, values }) => {
+                return (
+                    <Form>
+                        <DEDialog
+                            open={open}
+                            fullWidth={true}
+                            onClose={onClose}
+                            id={parentId}
+                            title="Details"
+                            actions={
+                                    <Button
+                                        id={build(parentId, ids.BUTTONS.OK)}
+                                        type="submit"
+                                        color="primary"
+                                    onClick={onClose}
+                                    >
+                                        {i18nCommon("ok")}
+                                </Button>
+                            }
+                        >
+                            {isDetailsFetching && (
+                                <Skeleton
+                                    animation="wave"
+                                    variant="rect"
+                                    height={400}
+                                />
+                            )}
+                            {detailsFetchError && (
+                                <ErrorTypographyWithDialog
+                                    errorObject={detailsFetchError}
+                                    errorMessage={t("requestDetailsFetchError")}
+                                />
+                            )}
+                            {!detailsFetchError && !isDetailsFetching && (
+                                <>
+                                    <Field
+                                        name="name"
+                                        label={t("toolName")}
+                                        id={build(
+                                            parentId,
+                                            ids.TOOL_REQUEST.NAME
+                                        )}
+                                        component={FormTextField}
+                                        InputProps={{
+                                            readOnly: true,
+                                        }}
+                                    />
```

Comment:
All of these form fields are always read-only, and Formik's `handleSubmit` function is never called, so I don't think these need to be in a form at all 🤔 

It seems like all of these `FormTextField` components can be replaced with a vanilla MUI `TextField` instead.
For example here:
```suggestion
                                    <TextField
                                        value={details.name}
                                        label={t("toolName")}
                                        id={build(
                                            parentId,
                                            ids.TOOL_REQUEST.NAME
                                        )}
                                        InputProps={{
                                            readOnly: true,
                                        }}
                                        variant="outlined"
                                        margin="dense"
                                        fullWidth
                                    />
```

Reply:
Good catch. But I guess, `Formik` provides `json` to form field initialization / value mapping and I don't see any performance issues associated with `Formik`.  So unless there is strong preference to not use `Formik`, I will leave it alone for now.